Source/WebCore/loader/soup/ResourceLoaderSoup.cpp

```diff
@@ -34,14 +34,45 @@
 #include "ResourceError.h"
 #include "SharedBuffer.h"
 #include <gio/gio.h>
+#include <optional>
+#include <wtf/SortedArrayMap.h>
 #include <wtf/glib/GRefPtr.h>
 #include <wtf/glib/GUniquePtr.h>
 #include <wtf/glib/RunLoopSourcePriority.h>
+#include <wtf/text/ASCIILiteral.h>
 #include <wtf/text/StringView.h>
 #include <wtf/text/WTFString.h>
 
 namespace WebCore {
 
+static std::optional<String> contentTypeLookUpForKnownResource(const char* filename)
+{
+    String fileName = String::fromLatin1(filename);
+    StringView extension;
+    size_t dotIndex = fileName.reverseFind('.');
+
+    if (dotIndex != notFound && dotIndex < (fileName.length() - 1))
+        extension = StringView(fileName).substring(dotIndex);
+
+    static constexpr std::pair<ComparableLettersLiteral, ASCIILiteral> extensionTypeMapping[] = {
```

Comment:
I think we should search case insensitive. Use ComparableCaseFoldingASCIILiteral instead of ComparableLettersLiteral?